src/serializer/ResidualHeapSerializer.js

```diff
@@ -1197,7 +1208,6 @@ export class ResidualHeapSerializer {
       },
     };
 
-    invariant(this.serializedValues.size === this.residualValues.size);
```

Comment:
That's an important invariant that I don't want to remove easily... Let's see what we can do.